retrieve-api/src/main/java/com/connexta/retrieve/rest/spring/RetrieveApi.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright (c) 2019 Connexta, LLC
+ *
+ * Released under the GNU Lesser General Public License version 3; see
+ * https://www.gnu.org/licenses/lgpl-3.0.html
+ */
+package com.connexta.retrieve.rest.spring;
+
+import org.springframework.core.io.Resource;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+
+public interface RetrieveApi {
+
+  @GetMapping(value = "/retrieve/{id}")
+  default ResponseEntity<Resource> retrieve(@PathVariable final String id) {
```

Comment:
I don't think we should return a ResponseEntity from an API. ResponseEntity is an implementation detail and returning from the method is a leaky abstraction. I think we should return the Resource.

Reply:
`ResponseEntity` isn't exposed beyond the endpoint. See the `RetrieveController#retrieve`. The ingest API defined by a yaml spec returns `ResponseEntity<Void>` (see https://github.com/connexta/ion-ingest-api/blob/ion-ingest-api-0.1.1/rest/specs/src/main/filtered-resources/openapi/openapi.yaml).

How would we return different status codes without using `ResponseEntity`? We also need to add a file name to the header so that the product is download with the original filename and extension. Feel free to make a new PR to fix it if you know how to fix these issues without using `ResponseEntity`.